### ISSUE TITLE
How useful is the /network/fees/transaction endpoint when it returns all zeros

### ISSUE THREAD

**Issue by Wayonb** [nemtech/catapult-rest#326]
When there is no transaction activity on the blockchain, the network/fees/transaction endpoint returns all zeros.  These values cannot be used on the testnet where the default minFeeMultiplier = 100 and transactions will get dropped.

{"averageFeeMultiplier":0,"medianFeeMultiplier":0,"highestFeeMultiplier":0,"lowestFeeMultiplier":0}

		server.get('/network/fees/transaction', (req, res, next) => {
			const numBlocksTransactionFeeStats = services.config.numBlocksTransactionFeeStats || 1;
			return db.latestBlocksFeeMultiplier(numBlocksTransactionFeeStats).then(feeMultipliers => {
				res.send({
					averageFeeMultiplier: Math.floor(average(feeMultipliers)),
					medianFeeMultiplier: Math.floor(median(feeMultipliers)),
					highestFeeMultiplier: Math.max(...feeMultipliers),
					lowestFeeMultiplier: Math.min(...feeMultipliers)
				});
				next();
			});
		});

**Comment by Vektrat** [comment]
i agree with this, however i do not think it is a bug, and it is working as expected, the documentation is very clear on it in here https://nemtech.github.io/symbol-openapi/v0.8.10/#operation/getTransactionFees 

> Returns the **average, median, highest and lower fee multiplier over the last "numBlocksTransactionFeeStats"**. The setting "numBlocksTransactionFeeStats" is adjustable via the configuration file (rest/resources/rest.json) per REST instance.

it's true that for no data, 0 is an invalid value, so maybe we should return `undefined`, but other than that, i don't know what could be changed to improve this endpoint

i suppose what would be "useful" would be to always return some kind of average, but as you can see, and per description, this is not always possible, so in my opinion clients would need to adapt to the case where this information may not always be possible, after all you are asking for historical data, it is not as simple as having fixed fees

@dgarcia360 thoughts?

**Comment by fboucquez** [comment]
My 2 cents

I guess having no block information is a pretty odd case right? Could the clients use network's properties [defaultDynamicFeeMultiplier](https://github.com/nemtech/symbol-openapi/blob/master/spec/core/network/schemas/ChainPropertiesDTO.yml#L48) for a proper default in that case?

**Comment by dgarcia360** [comment]
``When there is no transaction activity on the blockchain, the network/fees/transaction endpoint returns all zeros.``

I don't think not having transactions activity on the blockchain is the case, blocks are harvested anyway and they should contain as well the fee multiplier from the harvester node.

``cannot be used on the testnet where the default minFeeMultiplier = 100``

Note that minFeeMultiplier is a value configurable per node, not per network. If all the nodes configure the value to be 100, shouldn't be the feeMultiplier of every block 100, hence the average should be around 100?

The problem (haven't tested it) could be that the values returned are  0 because the REST API doesn't calculate the median if there are less than ``numBlocksTransactionFeeStats`` blocks harvested. In this case, the API could either return ``undefined`` or calculate the feeMultiplier with the number of blocks available < ``numBlocksTransactionFeeStats``.

**Comment by Vektrat** [comment]
>  In this case, the API could either return undefined or calculate the feeMultiplier with the number of blocks available < numBlocksTransactionFeeStats

i'd not be happy with adding a default/fallback behaviour to force a situation so that we can have a cute response - the `numBlocksTransactionFeeStats` is what it is, and the formula should always behave as described even when preconditions are not ideal - let's not forget this calculation is a best effort operation in order to give a hint to the user
